rclcpp/src/rclcpp/logger.cpp

```diff
@@ -44,4 +44,17 @@ get_node_logger(const rcl_node_t * node)
   return rclcpp::get_logger(logger_name);
 }
 
+void
+Logger::set_level(Level level)
+{
+  if (rcutils_logging_set_logger_level(
+      get_name(),
+      static_cast<RCUTILS_LOG_SEVERITY>(level)) != RCUTILS_RET_OK)
```

Comment:
instead of wrapping possible errors into single one, I would add if statement to let the user know why method fails with appropriate message.

Reply:
It seems reasonable. I'll update it.
(I found some errors in `rcutils`, such as `RCUTILS_RET_LOGGING_SEVERITY_MAP_INVALID`, isn't correspond to `rcl`, so I cast it with `RCL_RET_ERROR` only.)

Reply:
> I found some errors in rcutils, such as RCUTILS_RET_LOGGING_SEVERITY_MAP_INVALID, isn't correspond to rcl, so I cast it with RCL_RET_ERROR only.

i think that is okay. with using rclcpp::exceptions::throw_from_rcl_error, error should be changed into `rcl`